sympy/functions/elementary/complexes.py

```diff
@@ -58,6 +58,8 @@ def eval(cls, arg):
             return arg
         elif arg.is_imaginary or (S.ImaginaryUnit*arg).is_real:
             return S.Zero
+        elif arg.is_Matrix:
+            return arg._eval_real()
```

Comment:
I think it would be better to reuse the `as_real_imag()` method that Expr already uses.